docs/architecture.md

```diff
@@ -1,280 +1,46 @@
 # Architecture
 
-This document describes the internal components that the Flow Data Provisioning Service is constituted of, as well as
-the API it exposes.
+The following document gives a brief overview of different parts of the Flow DPS architecture.
 
-**Table of Contents**
+## Flow DPS Indexer
 
-1. [Chain](#chain)
-    1. [Disk Chain](#disk-chain)
-2. [Feeder](#feeder)
-    1. [Disk Feeder](#disk-feeder)
-3. [Mapping State Machine](#mapping-state-machine)
-4. [Index](#index)
-    1. [Database Schema](#database-schema)
-        1. [First Height](#first-height)
-        2. [Last Height](#last-height)
-        3. [Header Index](#header-index)
-        4. [Commit Index](#commit-index)
-        5. [Events Index](#events-index)
-        6. [Path Deltas Index](#path-deltas-index)
-        7. [Block Height Index](#block-height-index)
-        8. [Transaction Records](#transaction-records)
-        9. [Block Transaction Index](#block-transaction-index)
-        10. [Collection Index](#collection-index)
-        11. [Collection Guarantee Index](#collection-index)
-        12. [Block Collection Index](#block-collection-index)
-        13. [Transaction Result Index](#transaction-result-index)
-        14. [Seals Index](#seals-index)
-        15. [Block Seals Index](#block-seals-index)
-        16. [Transaction Height Index](#transaction-height-index)
+The Flow DPS Indexer can only index past sporks, contrary to the [Flow DPS Live](#flow-dps-live) binary which indexes live sporks.
+In order to build its index, it uses the spork's root checkpoint, and goes through the protocol state database and write-ahead logs from its nodes to index the state of the blockchain throughout its history.
+It does not run the DPS API on its own, so the API must be started using the produced index, once the indexing operation is complete.
 
-## Chain
+[![Flow DPS Indexer](./svg/flow_dps_indexer.svg)](./svg/flow_dps_indexer.svg)
 
-The Chain component is responsible for reconstructing a view of the sequence of blocks, along with their metadata.
-It allows the consumer to step from the root block to the last sealed block, while providing data related to each height along the sequence of blocks, such as block identifier, state commitment and events.
-It is used by the [Mapper](#mapping-state-machine) to map a set of deltas from the [Feeder](#feeder) to each block height.
+### Components
 
-[Package documentation](https://pkg.go.dev/github.com/optakt/flow-dps/service/chain)
+* [Loader](https://pkg.go.dev/github.com/optakt/flow-dps/service/loader) -- Loads checkpoints and retrieves their root trie.
+* [Feeder](https://pkg.go.dev/github.com/optakt/flow-dps/service/feeder) -- Loads trie updates in chronological order.
+* [Chain](https://pkg.go.dev/github.com/optakt/flow-dps/service/chain) -- Reads block data at any given height from the protocol state database.
```

Comment:
I would remove `at any given height` from this. Also, `chain data` is better than `block data`.